autoload/sj/js.vim

```diff
@@ -122,7 +122,7 @@ function! sj#js#JoinOneLineIf()
   let end_line_no = if_line_no + 2
   let end_line = getline(end_line_no)
 
-  if if_line !~ '^\s*if (.+) {' && end_line !~ '^\s*}\s*$'
+  if if_line !~ '^\s*if (.\+) {' || end_line !~ '^\s*}\s*$'
```

Comment:
Good job on noticing these :+1:. I guess this definition was pretty rudimentary.